csharp/src/Apache.Arrow.Flight/FlightInfo.cs

```diff
@@ -64,7 +64,9 @@ internal Protocol.FlightInfo ToProtocol()
             var response = new Protocol.FlightInfo()
             {
                 Schema = serializedSchema,
-                FlightDescriptor = Descriptor.ToProtocol()
+                FlightDescriptor = Descriptor.ToProtocol(),
+                TotalBytes = TotalBytes,
+                TotalRecords = TotalRecords
```

Comment:
This isn't related to your PR but should `totalRecords` and `totalBytes` default to -1 in the constructor?  From https://arrow.apache.org/docs/format/Flight.html it appears that `-1` is the value to use for "unknown".